cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteProcessController.java

```diff
@@ -40,4 +44,11 @@ public interface RemoteProcessController {
    * @throws Exception if not able to kill the remote process
    */
   void kill() throws Exception;
+
+  /**
+   * Kills the remote process depending on the ProgramStatus
+   * @throws Exception if not able to kill the remote process
+   * @param runtimeJobStatus
+   */
+  void kill(RuntimeJobStatus runtimeJobStatus) throws Exception;
```

Comment:
I think you can just modify the old kill method to take a status instead of adding a new method.

Reply:
Done.